docs/src/architecture/pallets/index.md

```diff
@@ -29,19 +29,18 @@ After publishing, the funds allocated for the deal will be moved from `free` to
 <img src="../../images/market/publish_storage_deals.svg" alt="Publishing storage deals">
 
 At this point, the remaining responsibility is shifted to the storage provider, which needs to activate the deal.
-First, the storage provider needs to call `get_randomness` from the [Randomness Pallet](./randomness.md) in order to create a replica and
+First, the storage provider uses randomness (via the [Randomness trait](./randomness.md) implementation) in order to create a replica and
 [pre-commit](./storage-provider.md#pre_commit_sectors) the deal's sectors.
-The sealing and pre-committing takes some time, after that the storage provider needs to fetch yet another randomness seed to create a proof.
-Subsequently, they [prove](./storage-provider.md#prove_commit_sectors) they stored the sectors by calling [`prove_commit_sectors`](./storage-provider.md#prove_commit_sectors) extrinsics.
+The sealing and pre-committing takes some time, after that the storage provider needs additional randomness to create a proof.
+Subsequently, they [prove](./storage-provider.md#prove_commit_sectors) they stored the sectors by calling the [`prove_commit_sectors`](./storage-provider.md#prove_commit_sectors) extrinsic.
 
 Verification is done via the [Proofs Pallet](./proofs.md) and reported to the Market pallet to terminate the deal and apply penalties to the storage provider
-(remove and burn its collateral — i.e. `locked` funds) if they fail to activate the deal on time and return the funds to the client.
+(remove and burn its collateral - i.e. `locked` funds) if they fail to activate the deal on time or properly maintain storage, the client's funds are returned.
```

Comment:
The em-dash was correct, it is being used as parentheses
https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use